src/net/socket_utils.zig

```diff
@@ -12,7 +12,7 @@ pub const PACKETS_PER_BATCH: usize = 64;
 
 pub fn readSocket(
     allocator: std.mem.Allocator,
-    socket: *UdpSocket,
+    socket_: UdpSocket,
```

Comment:
The main reason was that it complicates management of lifetimes. I think I was running into some segfaults.

When this function takes a pointer, you need to make sure the data it points to has a lifetime that outlives this function. Usually the data is created in one thread, and this runs in another thread. There is less to worry about if you pass it by value.

`Socket` is basically just a file descriptor and an ip address. It's 40 bytes of static data. It can be conceptualized as something like a fat pointer to an OS resource. So `*Socket` is like a pointer to a pointer, which is an unnecessary layer of indirection.